content/YubiHSM2/Usage_Guides/YubiHSM_2_for_Microsoft_SQL_Server_Deployment_Guide--Enabling_Always_Encrypted_with_YubiHSM_2/Introduction.adoc

```diff
@@ -0,0 +1,36 @@
+
+These guidelines for deployment cover basic topics, so the instructions should be modified as required for your specific environment.  It is assumed that you are familiar with the concepts and processes for working with Microsoft SQL Server. It is also assumed that the installation is performed on a single Microsoft SQL Server database, but the concept can be extended to more servers and databases.
+
+[IMPORTANT]
+===========
+Before deploying to production, we recommend that you install and test the Microsoft SQL Server installation and setup of the YubiHSM 2 in a test or lab environment. For guidance on enabling the Microsoft SQL Server feature Always Encrypted in a production environment, see https://docs.microsoft.com/en-us/sql/relational-databases/security/encryption/always-encrypted-database-engine?view=sql-server-2017[the Microsoft SQL Docs for Always Encrypted].
+===========
+
+
+=== Introduction to Always Encrypted
+
+Introduced in 2016, Microsoft SQL Server’s Always Encrypted feature enables the encryption of selected columns in a database.
+
+Note: The YubiHSM 2 requires Microsoft SQL Server 2017 and Microsoft SQL Server Management Studio (SSMS) 2018.
+
+The Always Encrypted encryption mechanisms rely upon two cryptographic keys, described in detail in the Microsoft SQL Docs, https://docs.microsoft.com/en-us/sql/relational-databases/security/encryption/overview-of-key-management-for-always-encrypted?view=sql-server-2017[Overview of Key Management for Always Encrypted]. In brief:
+
+* The Column Encryption Key (CEK) is a symmetric key used for encrypting the contents of a selected database column.
+* The Column Master Key (CMK) is an asymmetric key that is used for protecting the encryption key. The CMK for Always Encrypted can be protected in a local key store, which is in the scope of this document, or in a centralized key store, which is not in scope.
+
+A CMK can be stored in a local key store that supports Microsoft’s Cryptography Next Generation (CNG) API. In order to protect the CMK in hardware, a hardware security module (HSM) should be used. In this scenario, Always Encrypted accesses the HSM through the CNG API by using a key storage provider (KSP).
+
+In order to protect the CMK in hardware, the YubiHSM 2 can be deployed as the local key store. Microsoft’s Always Encrypted accesses the YubiHSM 2 through the KSP that is provided with the YubiHSM software tools. With this setup, the Microsoft SQL Server Management Studio (SSMS) can be used to manage the CMK in the YubiHSM 2 device. This deployment guide describes two ways to generate the CMK and CEK in YubiHSM 2:
+
+* By using the SSMS wizard, as described in link:Use_SMS_to_Generate_the_CMK_and_CEK[Use SSMS to Generate the CMK and CEK]
```

Comment:
this link is broken, the document seems to be named Use_SSMS_to_Generate_the_CMK_and_CEK.adoc

Reply:
6870c4b8b4c60f4a0e4971df7d646f8563a377d0